AP_HAL_ChibiOS: add hwdef X-MAV-AP-H743v2

TompsonTan opened the pull request:
Dear ArduPilot-Autopilot Developer, We are X-MAV, a new manufacturer of flight controllers. We have developed a flight controller AP-H743v2, We will keep developing it and want to add our flight controller to supported hardware.

Best regard!

![AP-H743v2_3D_1](https://github.com/user-attachments/assets/a16c6665-8b65-46b3-89e3-cd4fd6c85e59)

andyp1per commented on Tools/AP_Bootloader/board_types.txt:
```diff
@@ -392,6 +392,9 @@ AP_HW_CUAV-7-NANO                    7000
 AP_HW_VUAV-V7pro                     7100
 
+# IDs 7200-7299 reserved for X-MAV
+AP_HW_X-MAV-AP-H743V2                7200
```
need to reserve only 10 IDs here

TompsonTan replied:
ok, i will fix it

andyp1per commented on libraries/AP_HAL_ChibiOS/hwdef/X-MAV-AP-H743v2/defaults.parm:
```diff
@@ -0,0 +1,11 @@
+BATT_MONITOR 4
+
+#Serial Port defaults
+SERIAL6_PROTOCOL 42  #MSP OSD
```
You can move these into the hwdef by using

define DEFAULT_SERIALx_PROTOCOL yy

andyp1per commented on libraries/AP_HAL_ChibiOS/hwdef/X-MAV-AP-H743v2/defaults.parm:
```diff
@@ -0,0 +1,11 @@
+#BATTERY MONITOR ON
+BATT_MONITOR 4
```
Should be in hwdef

TompsonTan replied:
fixed

andyp1per reviewed (dismissed):
Few minor changes

andyp1per commented:
The commits need to be split across subsystems

TompsonTan commented:
@andyp1per Hi, can these commits be merged now? Please feel free to let me know if there is any need to change it.

andyp1per commented on libraries/AP_HAL_ChibiOS/hwdef/X-MAV-AP-H743v2/hwdef.dat:
```diff
@@ -165,6 +165,17 @@ COMPASS IST8310 I2C:ALL_INTERNAL:0x0E false ROTATION_YAW_90
+define BATT_MONITOR 4
+
+#Serial Port defaults
+define SERIAL6_PROTOCOL 42  #MSP OSD
```
DEFAULT_SERIAL6_PROTOCOLetc

peterbarker commented on Tools/AP_Bootloader/board_types.txt:
```diff
@@ -392,6 +392,9 @@ AP_HW_CUAV-7-NANO                    7000
 AP_HW_VUAV-V7pro                     7100
 
+# IDs 7200-7209 reserved for X-MAV
+AP_HW_X-MAV-AP-H743V2                7200
```
Please don't leave this huge gap in IDs.  Fill in one of our *other* huge gaps.

TompsonTan replied:
ok, can i use 7110-7119 ?

TompsonTan replied:
Already changed AP_HW_X-MAV-AP-H743V2 id to 1168, and filled the gap now.

TompsonTan commented:
1.changed reserved id;
2.added document readme.md

Please review. Thanks again!

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/X-MAV-AP-H743v2/hwdef.dat:
```diff
@@ -0,0 +1,183 @@
+define HAL_OS_FATFS_IO 1
+
+#Serial Port defaults
+define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_MAVLink2
```
remove...this is default

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/X-MAV-AP-H743v2/hwdef.dat:
```diff
@@ -0,0 +1,183 @@
+
+#Serial Port defaults
+define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_MAVLink2
+define DEFAULT_SERIAL3_PROTOCOL SerialProtocol_GPS
```
remove, this is default
instead :
 define DEFAULT_SERIAL4_PROTOCOL SerialProtocol_None
or it will be GPS2 by default

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/X-MAV-AP-H743v2/README.md:
```diff
@@ -0,0 +1,95 @@
+
+## RC Input
+
+The default RC input is configured on the UART5. The SBUS pin is inverted and connected to RX5. RC could  be applied instead at a different UART port such as UART2, UART4 or UART6 or UART8, and set the protocol to receive RC data: `SERIALn_PROTOCOL=23` and change SERIAL7_Protocol to something other than '23'.
```
```suggestion
The default RC input is configured on the UART5 and supports all RC protocols except PPM. The SBUS pin is inverted and connected to RX5. When using RX5 or SBUS, the other input should be unconnected. RC can be attached to any UART port as long as the serial port protocol is set to `SERIALn_PROTOCOL=23` and SERIAL7_Protocol  is changed to something other than '23'.

```

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/X-MAV-AP-H743v2/README.md:
```diff
@@ -0,0 +1,95 @@
+
+## OSD Support
+
+The AP-H743v2 supports onboard OSD using OSD_TYPE 1 (MAX7456 driver). Simultaneously, DisplayPort OSD is available on the HD VTX connector.
```
```suggestion
The AP-H743v2 supports onboard analog SD OSD using a MAX7456 chip. Simultaneously, DisplayPort HD OSD is available on the DJI connector for HD VTX. Both on board OSD and DisplayPort OSD can be operated simultaneously.

```

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/X-MAV-AP-H743v2/README.md:
```diff
@@ -0,0 +1,95 @@
+
+The AP-H743v2 supports up to 8 PWM outputs.
+
+All the channels support DShot.
```
```suggestion
All the channels support DShot and BiDir DShot.

```

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/X-MAV-AP-H743v2/README.md:
```diff
@@ -0,0 +1,95 @@
+
+All the channels support DShot.
+
+Channels 1-8 support bi-directional DShot.
```
remove

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/X-MAV-AP-H743v2/README.md:
```diff
@@ -0,0 +1,95 @@
+
+Channels 1-8 support bi-directional DShot.
+
+PWM outputs are grouped and every group must use the same output protocol:
```
```suggestion
Outputs are grouped and every output within a group must use the same output protocol:

```

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/X-MAV-AP-H743v2/README.md:
```diff
@@ -0,0 +1,95 @@
+
+Initial firmware load can be done with DFU by plugging in USB with the bootloader button pressed. Then you should load the "with_bl.hex" firmware, using your favorite DFU loading tool.
+
+Once the initial firmware is loaded you can update the firmware using any ArduPilot ground station software. Updates should be done with the "*.apj" firmware files.
```
```suggestion
Once the initial firmware is loaded you can update the firmware using any ArduPilot ground station software. Updates should be done with the "\*.apj" firmware files.

```

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/X-MAV-AP-H743v2/hwdef-bl.dat:
```diff
@@ -0,0 +1,43 @@
+# hw definition file for processing by chibios_hwdef.py
```
you might consider adding the SD card interface in the bootloader to allow loading firmware from the SD card

Hwurzburg commented:
once final changes made, commits should be squashed and a library split made....you might want to break out the build_types.txt in a separate PR to reserve the bd id so someone else does not get it (first come/first served)...it gets merged almost immediately while this PR might take a review cycle or 2

TompsonTan commented:
Done:
1.added SD card interface in the bootloader to allow loading firmware from the SD card.
2.changed serial protocol default setting.
3.updated README.

TompsonTan commented:
board id reservation has been separate into another pr [https://github.com/ArduPilot/ardupilot/pull/28253](url). Please merge it, thanks.

Hwurzburg commented:
Since this looks close, I squashed and split for you...also removed the double define of OSD_TYPE in the hwdef.dat
did you update the bootloader bin with the new board id?

TompsonTan commented:
bootloader of X-MAV-AP-H743v2 with the new board id 1174 updated now.

TompsonTan commented:
> Since this looks close, I squashed and split for you...also removed the double define of OSD_TYPE in the hwdef.dat did you update the bootloader bin with the new board id?

bootloader of X-MAV-AP-H743v2 with the new board id 1174 updated now.

Hwurzburg reviewed (approved):
Johnson, I squashed the new bootloader and marked for DEVCall
Would it be possible to get a couple of samples? I would like to update my DShot test fixture with an H7 board and this one looks ideal

TompsonTan commented:
> Johnson, I squashed the new bootloader and marked for DEVCall Would it be possible to get a couple of samples? I would like to update my DShot test fixture with an H7 board and this one looks ideal

Please send me an email and give me your delivery address and contact, i will send you 2 samples two weeks later.

TompsonTan commented:
> Johnson, I squashed the new bootloader and marked for DEVCall Would it be possible to get a couple of samples? I would like to update my DShot test fixture with an H7 board and this one looks ideal

ok, i will send you some samples later.
